### DISCUSSION TITLE
Clarification of forwarded headers (Cloudflare + Caddy)

### DISCUSSION THREAD

**Discussion by simsrw73** [authelia/authelia#3977] [Q&A] [answered]
Networking and internet tech have always been one of my weakest subjects. I'm trying to setup a home server where Cloudflare manages my DNS (may use cloudflared tunnel also), and Caddy sits on my server (Proxmox => Debian VM => Docker) running in a container. I want to integrate authentication with Authelia, but I'm having trouble understanding the docs on forwarded headers (docs/content/en/integration/proxies/fowarded-headers/index.md). 

1. Is this two different rules that I am supposed to create?
2. If yes, how do they work? I'm likely reading it wrong but doesn't the first rule delete the header anytime it is present while the second removes it if it is not in a list of trusted ips? Doesn't that make the second rule redundant?
3. What goes in the list of trusted ips? cloudflare.com/ips ?

Apologies for my ignorance, but I'm just having a hard time understanding what I'm supposed to be accomplishing at this step.

**Comment by james-d-elliott** [comment, marked as answer]
You create 1 of the rules. If you have proxies that sit on foreign networks you may want the `Remove When Not From Trusted Source` option, otherwise you want the `Always Remove` option.

If you configure `Remove When Not From Trusted Source`  you use the IP's of the foreign proxies. This effectively "trusts" these proxies to provide a secure X-Forwarded-For header.
